### PR TITLE
[DPE-2079] Update rockcraft using Pebble services

### PR DESCRIPTION
**Pull request by deusebio**
Before merging, the workaround based on Docker should go away, when the [PR on Rockcraft](https://github.com/canonical/rockcraft/issues/235) is merged. 

Nevertheless the new image (which is also tested against integration tests) seems to be working properly

### CODE REVIEW

**Review by juditnovak — APPROVED** [review]
I kinda assume that this is what we need to do, and this is probably the best way :-)

File: .github/workflows/build.yaml
Comment (deusebio) [inline .github/workflows/build.yaml]: **note** This is a step quite time consuming, but I don't see a reason to have this in, given in the integration tests we just upload things directly to the microk8s ctr

Can I just get rid of this? @taurus-forever

**Comment by deusebio** [comment]
The [PR on Rockcraft](https://github.com/canonical/rockcraft/issues/235) has been merged, and I have update this accordingly. The PR is now ready for review

File: .github/workflows/build.yaml
Comment (cjdcordeiro) [inline .github/workflows/build.yaml]: for future reference: there's a rockcraft action that might be useful: https://github.com/canonical/craft-actions/tree/main#rockcraft-pack

File: files/pebble/pebble-start.sh
Comment (cjdcordeiro) [inline files/pebble/pebble-start.sh]: I'm not 100% sure that running pebble from within pebble is recommended, but in any case, you don't actually need to do that right? this file's last line could simply be: `cd /opt/spark && /bin/bash /opt/spark/entrypoint.sh $*`

File: rockcraft.yaml
Comment (cjdcordeiro) [inline rockcraft.yaml:27]: considering my comment in https://github.com/canonical/charmed-spark-rock/pull/23/files#r1229032686, if you add `user` and `group` here, then your entrypoint.sh could possibly be reduced to a single line

**Review by cjdcordeiro — APPROVED** [review]
nice! I left a few comments that aren't really blocking this from being merged, but rather optimization hints. So it LGTM ;)

File: .github/workflows/build.yaml
Comment (taurus-forever) [reply .github/workflows/build.yaml]: You need it on the "release/publish" [stage](https://github.com/canonical/charmed-spark-rock/blob/3.3/edge/.github/workflows/publish.yaml#LL30C1-L33C1).

**Review by taurus-forever — CHANGES_REQUESTED** [review]
IMHO, you still need artifacts upload.

File: .github/workflows/build.yaml
Comment (taurus-forever) [inline .github/workflows/build.yaml:28]: Previously Jarred was changing version inside rockcraft.yaml:
```
yq e -i ".version = \"$tag\"" rockcraft.yaml
```
Are you sure you will have the same result/tag now?
LGTM, if so.

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**
